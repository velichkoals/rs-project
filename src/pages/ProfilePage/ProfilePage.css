.profilePage-wrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 3rem;
}

