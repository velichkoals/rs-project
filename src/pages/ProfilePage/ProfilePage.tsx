import React, { useState } from "react";
import "./ProfilePage.css";
import ProfileForm from "../../components/ProfileForm/ProfileForm";
import ProfileCard from "../../components/ProfileCard/ProfileCard";

export interface ICharacter {
  name: string;
  email: string;
  gender: string;
  correctBirthday: string;
  country: string;
  photo: string | Blob;
  allow: string;
  id: number;
}

const ProfilePage: React.FC = function () {
  const [array, setArray] = useState<Array<ICharacter>>([]);

  const handleFormValidation = (
    name: string,
    email: string,
    gender: string,
    correctBirthday: string,
    country: string,
    photo: string | Blob,
    allow: string,
    id: number
  ) => {
    const newCard = {
      name,
      email,
      gender,
      id,
      correctBirthday,
      country,
      photo,
      allow,
    };
    setArray([...array, newCard]);
  };
  return (
    <div className="profilePage-wrapper">
      <ProfileForm
        data-testid="profileForm"
        parentCallback={handleFormValidation}
      />
      {array.map((post, index) => (
        <ProfileCard key={index + 1} createCard={array[index]} />
      ))}
    </div>
  );
};

export default ProfilePage;
