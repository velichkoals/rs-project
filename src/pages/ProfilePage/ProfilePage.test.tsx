import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import ProfilePage from "./ProfilePage";
import React from "react";

it("ProfilePage renders", () => {
  const profilePage = render(
    <MemoryRouter>
      <ProfilePage />
    </MemoryRouter>
  );
  expect(profilePage).toMatchSnapshot();
});
