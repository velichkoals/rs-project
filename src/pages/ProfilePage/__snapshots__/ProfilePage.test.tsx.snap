// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profilePage-wrapper"
      >
        <div
          class="setup-form-wrapper"
          data-testid="form-wrapper"
        >
          <div
            class="setup-form__title"
          >
            <div
              class="setup-form__welcome"
            >
              Welcome back!
            </div>
            <div
              class="setup-form__setup"
            >
              Setup your profile
            </div>
          </div>
          <div
            class="form-wrapper"
          >
            <form
              class="form"
            >
              <div
                class="item__wrapper"
              >
                <div
                  class="form__item"
                >
                  <div
                    class="input-label"
                  >
                    Name
                  </div>
                  <input
                    class="form__item form__input"
                    id="name"
                    name="userName"
                    placeholder="Alexey"
                    type="text"
                  />
                </div>
                <div
                  class="form__error"
                />
              </div>
              <div
                class="item__wrapper"
              >
                <div
                  class="form__item"
                >
                  <div
                    class="input-label"
                  >
                    Email
                  </div>
                  <input
                    class="form__item form__input"
                    name="userEmail"
                    placeholder="mail@website.com"
                    type="text"
                  />
                </div>
                <div
                  class="form__error"
                />
              </div>
              <div
                class="item__wrapper"
              >
                <div
                  class="form__item"
                >
                  <div
                    class="input-label"
                  >
                    Select your gender
                  </div>
                  <div
                    class="form__item__gender-wrapper"
                  >
                    <div
                      class="form__item_gender"
                    >
                      <input
                        id="Male"
                        name="userGender"
                        type="radio"
                        value="Male"
                      />
                      <label
                        class="form__gender__label"
                        for="Male"
                      >
                        Male
                      </label>
                    </div>
                    <div
                      class="form__item_gender"
                    >
                      <input
                        id="Female"
                        name="userGender"
                        type="radio"
                        value="Female"
                      />
                      <label
                        class="form__gender__label"
                        for="Female"
                      >
                        Female
                      </label>
                    </div>
                  </div>
                </div>
                <div
                  class="form__error"
                />
              </div>
              <div
                class="item__wrapper"
              >
                <div
                  class="form__item"
                >
                  <div
                    class="input-label"
                  >
                    Date of birth
                  </div>
                  <div
                    class="form__input__date"
                  >
                    <input
                      class="form__input"
                      id="date-of-birth"
                      max="2024-01-01"
                      min="1900-01-01"
                      name="userBirthday"
                      placeholder="date of birth"
                      type="date"
                    />
                  </div>
                </div>
                <div
                  class="form__error"
                />
              </div>
              <div
                class="item__wrapper"
              >
                <div
                  class="form__item"
                >
                  <div
                    class="input-label"
                  >
                    Country
                  </div>
                  <select
                    class="form__item form__input form__input__country "
                    id="country"
                    name="userCountry"
                  >
                    <option
                      disabled=""
                      selected=""
                      value=""
                    >
                      Select your country
                    </option>
                    <option
                      value="Belarus"
                    >
                      Belarus
                    </option>
                    <option
                      value="Ukraine"
                    >
                      Ukraine
                    </option>
                    <option
                      value="Russia"
                    >
                      Russia
                    </option>
                  </select>
                </div>
                <div
                  class="form__error"
                />
              </div>
              <div
                class="item__wrapper"
              >
                <div
                  class="form__item"
                >
                  <div
                    class="input-label"
                  >
                    Upload your profile photo
                  </div>
                  <div>
                    <input
                      accept="image/png, image/gif, image/jpeg"
                      class="form__input form__input__file"
                      id="profile-photo"
                      name="userPhoto"
                      type="file"
                    />
                  </div>
                </div>
                <div
                  class="form__error"
                />
              </div>
              <div>
                <div
                  class="form__item"
                >
                  <input
                    class="form__item__checkbox"
                    id="receive-emails"
                    name="userAllow"
                    type="checkbox"
                  />
                  <label
                    class="input-label input__checkbox__label"
                    for="receive-emails"
                  >
                    Allow to receive emails
                  </label>
                </div>
              </div>
              <div
                class="form__item form__item__button-wrapper"
              >
                <input
                  class="form__item__button"
                  disabled=""
                  type="submit"
                  value="Create profile"
                />
              </div>
            </form>
            <div
              class="form__success"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="profilePage-wrapper"
    >
      <div
        class="setup-form-wrapper"
        data-testid="form-wrapper"
      >
        <div
          class="setup-form__title"
        >
          <div
            class="setup-form__welcome"
          >
            Welcome back!
          </div>
          <div
            class="setup-form__setup"
          >
            Setup your profile
          </div>
        </div>
        <div
          class="form-wrapper"
        >
          <form
            class="form"
          >
            <div
              class="item__wrapper"
            >
              <div
                class="form__item"
              >
                <div
                  class="input-label"
                >
                  Name
                </div>
                <input
                  class="form__item form__input"
                  id="name"
                  name="userName"
                  placeholder="Alexey"
                  type="text"
                />
              </div>
              <div
                class="form__error"
              />
            </div>
            <div
              class="item__wrapper"
            >
              <div
                class="form__item"
              >
                <div
                  class="input-label"
                >
                  Email
                </div>
                <input
                  class="form__item form__input"
                  name="userEmail"
                  placeholder="mail@website.com"
                  type="text"
                />
              </div>
              <div
                class="form__error"
              />
            </div>
            <div
              class="item__wrapper"
            >
              <div
                class="form__item"
              >
                <div
                  class="input-label"
                >
                  Select your gender
                </div>
                <div
                  class="form__item__gender-wrapper"
                >
                  <div
                    class="form__item_gender"
                  >
                    <input
                      id="Male"
                      name="userGender"
                      type="radio"
                      value="Male"
                    />
                    <label
                      class="form__gender__label"
                      for="Male"
                    >
                      Male
                    </label>
                  </div>
                  <div
                    class="form__item_gender"
                  >
                    <input
                      id="Female"
                      name="userGender"
                      type="radio"
                      value="Female"
                    />
                    <label
                      class="form__gender__label"
                      for="Female"
                    >
                      Female
                    </label>
                  </div>
                </div>
              </div>
              <div
                class="form__error"
              />
            </div>
            <div
              class="item__wrapper"
            >
              <div
                class="form__item"
              >
                <div
                  class="input-label"
                >
                  Date of birth
                </div>
                <div
                  class="form__input__date"
                >
                  <input
                    class="form__input"
                    id="date-of-birth"
                    max="2024-01-01"
                    min="1900-01-01"
                    name="userBirthday"
                    placeholder="date of birth"
                    type="date"
                  />
                </div>
              </div>
              <div
                class="form__error"
              />
            </div>
            <div
              class="item__wrapper"
            >
              <div
                class="form__item"
              >
                <div
                  class="input-label"
                >
                  Country
                </div>
                <select
                  class="form__item form__input form__input__country "
                  id="country"
                  name="userCountry"
                >
                  <option
                    disabled=""
                    selected=""
                    value=""
                  >
                    Select your country
                  </option>
                  <option
                    value="Belarus"
                  >
                    Belarus
                  </option>
                  <option
                    value="Ukraine"
                  >
                    Ukraine
                  </option>
                  <option
                    value="Russia"
                  >
                    Russia
                  </option>
                </select>
              </div>
              <div
                class="form__error"
              />
            </div>
            <div
              class="item__wrapper"
            >
              <div
                class="form__item"
              >
                <div
                  class="input-label"
                >
                  Upload your profile photo
                </div>
                <div>
                  <input
                    accept="image/png, image/gif, image/jpeg"
                    class="form__input form__input__file"
                    id="profile-photo"
                    name="userPhoto"
                    type="file"
                  />
                </div>
              </div>
              <div
                class="form__error"
              />
            </div>
            <div>
              <div
                class="form__item"
              >
                <input
                  class="form__item__checkbox"
                  id="receive-emails"
                  name="userAllow"
                  type="checkbox"
                />
                <label
                  class="input-label input__checkbox__label"
                  for="receive-emails"
                >
                  Allow to receive emails
                </label>
              </div>
            </div>
            <div
              class="form__item form__item__button-wrapper"
            >
              <input
                class="form__item__button"
                disabled=""
                type="submit"
                value="Create profile"
              />
            </div>
          </form>
          <div
            class="form__success"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
